stage1/init/init.go

```diff
@@ -344,6 +368,36 @@ func stage1() int {
 	return 0
 }
 
+func runsFromUnitFile() (bool, error) {
+	handle := C.dlopen(C.CString("libsystemd-login.so"), C.RTLD_LAZY)
+	if handle == nil {
+		// we can't open libsystemd-login.so so we assume systemd is not
+		// installed and we're not running from a unit file
+		return false, nil
+	}
+	sd_pid_get_owner_uid := C.dlsym(handle, C.CString("sd_pid_get_owner_uid"))
+	if sd_pid_get_owner_uid == nil {
+		return false, fmt.Errorf("error resolving sd_pid_get_owner_uid function")
+	}
+
+	var uid C.uid_t
+	err := C.my_sd_pid_get_owner_uid(sd_pid_get_owner_uid, 0, &uid)
+
+	if r := C.dlclose(handle); r != 0 {
```

Comment:
don't we need to do this as a defer or something as we will miss the close if ` sd_pid_get_owner_uid == nil`